### PR TITLE
Enable task factories for .NET Core

### PR DESCRIPTION
**Pull request by jeffkl**
1. Allow task factories to load on .NET Core
2. Update TaskEngineAssemblyResolver for .NET Core
3. Log an error for CodeTaskFactory and XamlTaskFactory when running on .NET Core
4. Update unit tests

Related to #304

### CODE REVIEW

**Comment by jeffkl** [comment]
@dotnet-bot test OSX Build for CoreCLR please

**Comment by AndyGerlicher** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR please

**Comment by jeffkl** [comment]
@rainersigwald could you please review this today?

File: src/Build/Instance/TaskRegistry.cs [hunk 1]
```diff
@@ -1361,7 +1358,9 @@ private bool GetTaskFactory(TargetLoggingContext targetLoggingContext, ElementLo
                                 }
                                 finally
                                 {
+#if FEATURE_APPDOMAIN
```
Comment (rainersigwald) [inline src/Build/Instance/TaskRegistry.cs]: Not sure if it actually matters, but should this be outside the `finally` so the compiler/runtime doesn't have to build up infrastructure for finally?

File: src/Shared/TaskEngineAssemblyResolver.cs [hunk 2]
```diff
@@ -95,7 +117,16 @@ internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
 #else
                             return Assembly.LoadFrom(_taskAssemblyFile);
 #endif
+
+                        }
+#else
+                        AssemblyNameExtension taskAssemblyName = new AssemblyNameExtension(AssemblyLoadContext.GetAssemblyName(_taskAssemblyFile));
+                        AssemblyNameExtension argAssemblyName = new AssemblyNameExtension(assemblyName);
```
Comment (jeffkl) [inline src/Shared/TaskEngineAssemblyResolver.cs]: I have no idea, its basically a copy paste from above but uses `AssemblyLoadContext` where applicable.

File: src/Build/Instance/TaskRegistry.cs (hunk 1 above)
Comment (jeffkl) [inline src/Build/Instance/TaskRegistry.cs]: No the `taskFactoryLoggingHost` is still needed but the `MarkAsInactive()` method is not compiled into the .NET Core MSBuild.

**Comment by jeffkl** [comment]
@dotnet-bot test Windows_NT Build for Desktop please

**Comment by jeffkl** [comment]
@dotnet-bot test Windows_NT Build for Desktop please

**Comment by jeffkl** [comment]
@dotnet-bot test Windows_NT Build for Desktop please

**Pull request merged**